### PR TITLE
Pinned react (and co) to 16.0.0-alpha.4

### PR DESCRIPTION
**Pull request by bvaughn**
Some time ago (24a7665) the version of React specified in `package.json` was relaxed (presumably unintentionally). As a result, a recent backwards-incompatible release of React 16 alpha 5 caused some breakage in OSS.

This PR pins the version of React to avoid breaking changes in subsequent alpha releases until the React team has created codemods and prepared the ReactNative codebase for this.

Since the change to `package.json` was made in February, this PR will need to go into the RC branch for the next ReactNative release in order to avoid breaking other apps in the OSS community.

cc @mkonicek, @gaearon, @javache

### CODE REVIEW

**Comment by bvaughn** [comment]
Why is CI trying to install `react-test-renderer@16.0.0-alpha.4` ? This version doesn't exist. That's why the `package.json` points at `16.0.0-alpha.3`

**Comment by janicduplessis** [comment]
@bvaughn Looks like the script for react-native init assumes that it all use the same version https://github.com/facebook/react-native/blob/master/local-cli/init/init.js#L91.

**Comment by bvaughn** [comment]
Gah! Thanks for pointing that out @janicduplessis

Updated that script as well.

File: local-cli/init/init.js [hunk 1]
```diff
@@ -60,10 +60,12 @@ function init(projectDir, argsOrName) {
  */
 function generateProject(destinationRoot, newProjectName, options) {
   var reactNativePackageJson = require('../../package.json');
-  var { peerDependencies } = reactNativePackageJson;
+  var { devDependencies, peerDependencies } = reactNativePackageJson;
   if (!peerDependencies) {
     console.error('Missing React peer dependency in React Native\'s package.json. Aborting.');
     return;
+  } else if (!devDependencies) {
+    console.error('Missing react-test-renderer dev dependency in React Native\'s package.json. Aborting.');
```
Comment (janicduplessis) [inline local-cli/init/init.js]: Should return here

File: local-cli/init/init.js [hunk 2]
```diff
@@ -72,6 +74,12 @@ function generateProject(destinationRoot, newProjectName, options) {
     return;
   }
 
+  var reactTestRendererVersion = devDependencies['react-test-renderer'];
+  if (!reactVersion) {
```
Comment (janicduplessis) [inline local-cli/init/init.js]: This should be `reactTestRendererVersion`

File: local-cli/init/init.js (hunk 1 above)
Comment (janicduplessis) [inline local-cli/init/init.js]: Also maybe update the message to `Missing dev dependencies ...

File: local-cli/init/init.js (hunk 2 above)
Comment (bvaughn) [inline local-cli/init/init.js]: Blah. Yeah, sorry. Multi-tasking fail. Will fix. :)

File: local-cli/init/init.js (hunk 1 above)
Comment (bvaughn) [inline local-cli/init/init.js]: Just copied above wording/formatting.

**Comment by janicduplessis** [comment]
@facebook-github-bot shipit

**Comment by bvaughn** [comment]
Thanks for the review, @janicduplessis!

**Comment by hramos** [comment]
Closing as these dependencies were recently updated to alpha6.

**Comment by janicduplessis** [comment]
Shoud we still ship the change in `local-cli/init/init.js` in case versions of react and test-utils don't match?

**Comment by hramos** [comment]
@bvaughn ?

**Comment by bvaughn** [comment]
Unsure. There was a transient problem with `16.0.0-alpha.4` because whoever published it did not also publish `react-test-renderer@16.0.0-alpha.6` which caused problems with the CI. Maybe this won't happen again? Hopefully not.

**Pull request closed without merging**